app/services/learning_units/resource_sorter_service.rb

```diff
@@ -0,0 +1,25 @@
+module LearningUnits
+  class ResourceSorterService
+    delegate :name, to: :learning_unit, prefix: :learning_unit
+    def initialize(learning_unit)
+      @learning_unit = learning_unit
+    end
+
+    def sorted_resources
+      resources = set_resources
+      resources.map do |array|
+        { id: array[0][0], name: array[0][1], evaluation: array[1] }
+      end
+    end
+
+    private
+
+    def set_resources
```

Comment:
I would choose a more descriptive name for the method.  I would suggest something like get_resources_sorted_by_evaluation